### PR TITLE
PLAT-850: isolate message sending from record building for SubSMRegister

### PR DESCRIPTION
**Pull request by bigbes**
also PLAT-849: isolate registration context

### CODE REVIEW

File: ledger-core/virtual/execute/execute.go [hunk 1]
```diff
@@ -948,7 +959,7 @@ func (s *SMExecute) stepWaitSafeAnswersRelease(ctx smachine.ExecutionContext) sm
 
 func (s *SMExecute) stepWaitSafeAnswers(ctx smachine.ExecutionContext) smachine.StateUpdate {
 	// waiting for all save responses to be there
-	stateUpdate := shared.CounterAwaitZero(ctx, s.lmnSafeResponseCounterLink)
+	stateUpdate := s.lmnContext.AwaitZeroSafeResponse(ctx)
```
Comment (cyraxred) [inline ledger-core/virtual/execute/execute.go:962]: may be WaitForAllSafeResponses?

File: ledger-core/virtual/execute/execute.go [hunk 2]
```diff
@@ -73,6 +72,7 @@ type SMExecute struct {
 	authenticationService authentication.Service
 	globalSemaphore       tool.RunnerLimiter
 	memoryCache           memoryCacheAdapter.MemoryCache
+	referenceBuilder      lmn.RecordReferenceBuilderService
```
Comment (cyraxred) [inline ledger-core/virtual/execute/execute.go:75]: is it really a service?

File: ledger-core/virtual/lmn/register.go [hunk 3]
```diff
@@ -106,178 +45,54 @@ type SubSMRegister struct {
...
-
-func (s *SubSMRegister) GetSubroutineInitState(ctx smachine.SubroutineStartContext) smachine.InitFunc {
-	ctx.SetSubroutineCleanupMode(smachine.SubroutineCleanupAliasesAndShares)
-
-	return s.Init
-}
-
-func (s *SubSMRegister) GetStateMachineDeclaration() smachine.StateMachineDeclaration {
-	return dSubSMRegisterInstance
+	PulseSlot        *conveyor.PulseSlot
```
Comment (cyraxred) [inline ledger-core/virtual/lmn/register.go:61]: I'd suggest to isolate this builder from PulseSlot due to unnecessary specifics.

File: ledger-core/virtual/lmn/register.go [hunk 4]
```diff
@@ -8,23 +8,15 @@
 package lmn
```
Comment (cyraxred) [inline ledger-core/virtual/lmn/register.go:8]: package name can be confusing - may be "vnlmn"?

File: ledger-core/virtual/execute/execute.go (hunk 1 above)
Comment (bigbes) [reply ledger-core/virtual/execute/execute.go]: fixed

File: ledger-core/virtual/lmn/register.go (hunk 3 above)
Comment (bigbes) [reply ledger-core/virtual/lmn/register.go]: fixed

File: ledger-core/virtual/execute/execute.go (hunk 2 above)
Comment (bigbes) [reply ledger-core/virtual/execute/execute.go]: fixed

File: ledger-core/virtual/lmn/register.go (hunk 4 above)
Comment (bigbes) [reply ledger-core/virtual/lmn/register.go]: fixed

**Review by cyraxred — APPROVED** [review]

**Review by MikhailZhukov — APPROVED** [review]

**Pull request merged**